### PR TITLE
tests: add new futex test

### PR DESCRIPTION
**Pull request by maksimmasalski**
Add new standalone futex test that verifies next requirements:
-Futex can be placed in user memory using ZTEST_BMEM
-User thread can write to futex value
-User threads can make wait/wake syscalls on it

Added detailed Doxygen tag with information about the test

Signed-off-by: Maksim Masalski <maksim.masalski@intel.com>

### CODE REVIEW

**Comment by maksimmasalski** [comment]
@nashif @andrewboie @andyross I'm looking for code review of that PR.

**Comment by maksimmasalski** [comment]
@smanni I find out that you added support for the waveshare_open103z, please help me to find out why build failed for your board. Is it a bug? Please open buildkite/zephyr webpage. In brief error is: /workdir/zephyr/boards/arm/waveshare_open103z/waveshare_open103z_defconfig:27: warning: attempt to assign the value 'n' to the undefined symbol CLOCK_STM32_PLL_XTPRE

**Comment by smanni** [comment]
> @smanni I find out that you added support for the waveshare_open103z, please help me to find out why build failed for your board. Is it a bug? Please open buildkite/zephyr webpage. In brief error is: `/workdir/zephyr/boards/arm/waveshare_open103z/waveshare_open103z_defconfig:27: warning: attempt to assign the value 'n' to the undefined symbol CLOCK_STM32_PLL_XTPRE`

CLOCK_STM32_PLL_XTPRE was already there before I started working on this new board. Don't you have the same warning even with other boards using the same kconfig (e.g. nucleo_f103rb)

**Comment by maksimmasalski** [comment]
I have issue that build can't pass for the waveshare_open103z board. Reported a bug #26700

**Review by wentongwu — APPROVED** [review]

File: tests/kernel/mem_protect/futex/src/main.c [hunk 1]
```diff
@@ -452,6 +451,81 @@ void test_user_futex_bad(void)
...
+ * @see atomic_set(), atomic_sub(), k_futex_wake(), k_futex_wait()
+ *
+ * @ingroup kernel_futex_tests
+ */
+void test_futex_locate_access(void)
+{
+
+	atomic_set(&simple_futex.val, 13);
+
+	k_thread_create(&futex_tid, stack_1, STACK_SIZE,
```
Comment (andrewboie) [inline tests/kernel/mem_protect/futex/src/main.c:509]: why are you doing this in a child thread?
couldn't this test have been defined using ztest_user_unit_test?

Comment (maksimmasalski) [reply tests/kernel/mem_protect/futex/src/main.c:510]: I did changes in pushed new commit that runs that test using ztest_user_unit_test

**Review by LeiW000 — APPROVED** [review]
It looks good to me.

File: tests/kernel/mem_protect/futex/src/main.c (hunk 1 above)
Comment (andrewboie) [reply tests/kernel/mem_protect/futex/src/main.c:510]: well I think you don't need a child thread, unless im missing something it doesn't seem like it is synchronizing with any other threads, it just runs and does its own thing. can't we move this logic up to the body of this function, and run it in user mode?

**Pull request merged**